java/wavefrontsender.md

```diff
@@ -0,0 +1,212 @@
+>Before your application can use a `WavefrontProxyClient`, you must [set up and start a Wavefront proxy](https://github.com/wavefrontHQ/java/tree/master/proxy#set-up-a-wavefront-proxy).
+
+When sending data via the Wavefront proxy, you need to create a `WavefrontProxyClient`. You need to build it with the information needed to send data to a Wavefront proxy, including:
+
+* The name of the host that will run the Wavefront proxy.
+* One or more proxy listening ports to send data to. The ports you specify depend on the kinds of data you want to send (metrics, histograms, and/or trace data). You must specify at least one listener port. 
+* Optional settings for tuning communication with the proxy.
+
+
+```java
+// Create the builder with the proxy hostname or address
+WavefrontProxyClient.Builder wfProxyClientBuilder = new WavefrontProxyClient.Builder(proxyHostName);
+
+// Set the proxy port to send metrics to. Default: 2878
+wfProxyClientBuilder.metricsPort(2878);
+
+// Set a proxy port to send histograms to.  Recommended: 2878
+wfProxyClientBuilder.distributionPort(2878);
+
+// Set a proxy port to send trace data to. Recommended: 30000
+wfProxyClientBuilder.tracingPort(30_000);
+
+// Optional: Set a custom socketFactory to override the default SocketFactory
+wfProxyClientBuilder.socketFactory(<SocketFactory>);
+
+// Optional: Set a nondefault interval (in seconds) for flushing data from the sender to the proxy. Default: 5 seconds
+wfProxyClientBuilder.flushIntervalSeconds(2);
+
+// Create the WavefrontProxyClient
+WavefrontSender wavefrontSender = wfProxyClientBuilder.build();
+ ```
+**Note:** When you [set up a Wavefront proxy](https://github.com/wavefrontHQ/java/tree/master/proxy#set-up-a-wavefront-proxy) on the specified proxy host, you specify the port it will listen to for each type of data to be sent. The `WavefrontProxyClient` must send data to the same ports that the Wavefront proxy listens to. Consequently, the port-related builder methods must specify the same port numbers as the corresponding proxy configuration properties: 
+
+| `WavefrontProxyClient` builder method | Corresponding property in `wavefront.conf` |
+| ----- | -------- |
+| `metricsPort()` | `pushListenerPorts=` |
+| `distributionPort()` | `histogramDistListenerPorts=` |
+| `tracingPort()` | `traceListenerPorts=` |
+ 
+
+## Option 2: Sending Data via Direct ingestion
+
+when sending data via direct ingestion, you need to create a `WavefrontDirectIngestionClient`, and build it with the information it needs to send data directly to Wavefront.
+
+### Step 1. Obtain Wavefront Access Information
+Gather the following access information:
+
+* Identify the URL of your Wavefront instance. This is the URL you connect to when you log in to Wavefront, typically something like `https://<domain>.wavefront.com`.
+* In Wavefront, verify that you have Direct Data Ingestion permission, and [obtain an API token](http://docs.wavefront.com/wavefront_api.html#generating-an-api-token).
```

Comment:
These are 2 completely different things. Should not be 1 bullet.

Reply:
Ack! Added permission as a prerequisite for step 1.